### PR TITLE
Ensure that Jax float0 array is recognized

### PR DESCRIPTION
**Pull request by NeilGirdhar**
See https://github.com/google/jax/issues/20620.

### CODE REVIEW

**Comment by NeilGirdhar** [comment]
This is intended to be a temporary solution until Jax (ideally) makes these arrays into Jax arrays.

**Comment by NeilGirdhar** [comment]
(Error appears to be unrelated to this pull request.)

**Comment by asmeurer** [comment]
This looks good. Can you add a test.

**Comment by NeilGirdhar** [comment]
Great, I added a test.  If you want, I could also add:
```python
assert isinstance(jax_zero, np.ndarray)  # If this triggers, the workaround can probably be removed
```

File: array_api_compat/common/_helpers.py [hunk 1]
```diff
@@ -18,6 +18,20 @@
 import inspect
 import warnings
 
+def is_jax_zero_gradient_array(x):
```
Comment (asmeurer) [inline array_api_compat/common/_helpers.py:21]:
```suggestion
def _is_jax_zero_gradient_array(x):
```

File: array_api_compat/common/_helpers.py [hunk 2]
```diff
@@ -44,7 +58,8 @@ def is_numpy_array(x):
     import numpy as np
 
     # TODO: Should we reject ndarray subclasses?
-    return isinstance(x, (np.ndarray, np.generic))
+    return (isinstance(x, (np.ndarray, np.generic))
+            and not is_jax_zero_gradient_array(x))
```
Comment (asmeurer) [inline array_api_compat/common/_helpers.py:62]:
```suggestion
            and not _is_jax_zero_gradient_array(x))
```

File: array_api_compat/common/_helpers.py [hunk 3]
```diff
@@ -149,7 +164,7 @@ def is_jax_array(x):
 
     import jax
 
-    return isinstance(x, jax.Array)
+    return isinstance(x, jax.Array) or is_jax_zero_gradient_array(x)
```
Comment (asmeurer) [inline array_api_compat/common/_helpers.py:167]:
```suggestion
    return isinstance(x, jax.Array) or _is_jax_zero_gradient_array(x)
```

**Comment by asmeurer** [comment]
The tests failure should be fixed by my suggestion, and the test suite failure should be fixed by https://github.com/data-apis/array-api-compat/pull/131.

**Comment by NeilGirdhar** [comment]
(Done.)

**Pull request merged**